command/agent/config/config.go

```diff
@@ -350,6 +351,140 @@ func (c *Config) ValidateConfig() error {
 		return fmt.Errorf("no auto_auth, cache, or listener block found in config")
 	}
 
+	return c.validateEnvTemplateConfig()
+}
+
+func (c *Config) validateEnvTemplateConfig() error {
+	// if we are not in env-template mode, exit early
+	if c.Exec == nil && len(c.EnvTemplates) == 0 {
+		return nil
+	}
+
+	if c.Exec == nil {
+		return fmt.Errorf("'exec' element must be specified with 'env_template' entries")
+	}
+
+	if len(c.EnvTemplates) == 0 {
+		return fmt.Errorf("must specify at least one 'env_template' element with a top-level 'exec' element")
+	}
+
+	if c.ExitAfterAuth {
+		return fmt.Errorf("'exit_after_auth' cannot be specified with 'env_template' entries")
```

Comment:
Why not? I get that this is a little weird for exec, but it's weird in the other cases, too. It could be used to test your config beforehand. In other words, since we don't restrict this for file templates, it seems weird to restrict it for env templates.

Reply:
Ah, interesting, did not know that. I suppose we could allow it then 👍